scheduler/pkg/agent/client_utils.go

```diff
@@ -112,7 +111,7 @@ func (b *backOffWithMaxCount) Reset() {
 }
 
 func (b *backOffWithMaxCount) NextBackOff() time.Duration {
-	if b.currentCount >= b.maxCount {
+	if b.currentCount >= b.maxCount-1 {
```

Comment:
Just to confirm: does this now retry once if b.maxCount is one? or is it that we now consider the initial function call as part of the "maxCount"?

Reply:
I guess the question is whether for a config parameter named something like (`max*RetryCount`) one would expect a maximum number of `max*RetryCount` retries, or they would expect that the function runs a total of `max*RetryCount` times.

Reply:
I added a ticket to create docs to clarify these semantic differences